src/main/java/io/aiven/kafka/connect/common/grouper/TopicPartitionRecordGrouper.java

```diff
@@ -66,20 +66,21 @@
         Objects.requireNonNull(filenameTemplate, "filenameTemplate cannot be null");
         Objects.requireNonNull(tsSource, "tsSource cannot be null");
         this.filenameTemplate = filenameTemplate;
-        this.setTimestampBasedOnRecord = record -> new Function<>() {
-            private final Map<String, DateTimeFormatter> timestampFormatters =
+
+        final Map<String, DateTimeFormatter> timestampFormatters =
                 Map.of(
-                    "yyyy", DateTimeFormatter.ofPattern("yyyy"),
-                    "MM", DateTimeFormatter.ofPattern("MM"),
-                    "dd", DateTimeFormatter.ofPattern("dd"),
-                    "HH", DateTimeFormatter.ofPattern("HH")
+                        "yyyy", DateTimeFormatter.ofPattern("yyyy"),
+                        "MM", DateTimeFormatter.ofPattern("MM"),
+                        "dd", DateTimeFormatter.ofPattern("dd"),
+                        "HH", DateTimeFormatter.ofPattern("HH")
                 );
-
+        this.setTimestampBasedOnRecord = record -> new Function<>() {
```

Comment:
Nit pick: Why not one more lambda? To me 
`record -> parameter -> tsSource.time(record).format(timestampFormatters.get(parameter.value()))` looks a bit cleaner but maybe it's just me biased by syntax of other languages with higher order functions support...

Reply:
Yeah, looks fine, did this